### ISSUE TITLE
Use transclude scope for inline node templates

### ISSUE THREAD

**Issue by jtrussell** [iVantage/angular-ivh-treeview#81]
I'd like to see what it would take to get rid of the need for wrapping transcluded node templates in a `script` tag so that folks can use functions/variables in the transcluded scope within their templates.

E.g. this:

```
<div ivh-treeview="fancy.treeData">
   <div custom-node-template scope-var="fancy.hooray"></div>
</div>
```

Instead of this:

```
<div ivh-treeview="fancy.treeData">
  <script type="text/ng-template">
    <div custom-node-template scope-var=":'("></div>
  </script>
</div>
```

Currently the former will generate a number of compilation errors.

See #70

**Comment by christophercr** [comment]
Hi, I have player around with the node templates and the tree compilation, and I haven't found a way to get rid of the wrapper of the transcluded templates (either with a script tag or with ng-include)

On the other hand, I could finally find a solution to use the transcluded scope in the node templates, obviously, because of the script or ng-include wrapper mentioned above, the way to have access to the transcluded scope is with $parent, like this (following the example of your comment):

`<div ivh-treeview="fancy.treeData">
  <script type="text/ng-template">
    <div custom-node-template scope-var="$parent.fancy.hooray"></div>
  </script>
</div>`

**Comment by jtrussell** [comment]
That's interesting. I wouldn't want to advocate using `$parent` in templates directly but if we can transparently expose that scope to the template I'd be on board :).

**Comment by christophercr** [comment]
Yes, the transcluded scope is transparently exposed (because the new scope prototypically inherits from that).

And regarding the `$parent`, I know, I don't feel 100% comfortable using it neither but this is due to the `script `or `ng-include` wrapping the template. As soon as I try to remove it, there is a compilation error due to the ivhTreeview directive being required, maybe something to do with the order in which the directives are compiled.

If we find a way to get rid of such wrapper then it would be great but so far this is the best I could do.

**Comment by christophercr** [comment]
Hi, do you think this feature could also be part of the v2? If so, I can submit the corresponding PR.

**Comment by jtrussell** [comment]
I'd love to get this in v2, we'd just want to make sure we got appropriate tests added as well.
